### ISSUE TITLE
migration failed when update to latest ghost

### ISSUE THREAD

**Issue by sooxt98** [TryGhost/Ghost#12129]
![image](https://user-images.githubusercontent.com/13378059/90176041-f5858a00-ddda-11ea-9039-e2cad9be54e7.png)

**Comment by sooxt98** [comment]
I tried to delete the `ghost_prod.members_stripe_customers_subscriptions` table, now i can't even rollback...

**Comment by matthanley** [comment]
Hey @sooxt98 👋 

In future, please use the issue template provided to ensure we can reproduce and effectively troubleshoot the issue.

For this issue, it would be great if you could provide:
* Reproduction steps
* Ghost Version (migrating _from_ and _to_)
* Node Version
* OS
* Installation method (e.g. manual/ghost-cli)

We also ask that you please do not use GitHub for help or support 😄. If you need assistance with your self-hosted Ghost instance, our [forum](https://forum.ghost.org/) is a great place to get community support, plus it helps create a central location for searching problems/solutions.

Thanks!

**Comment by sooxt98** [comment]
@matthanley 

Reproduction steps
cd /var/www/ghost
ghost update

Ghost Version (migrating from and to)
2 to 3 (not sure 2.xxx which)

Node Version
12.18.3

OS
ubuntu

installation method
ghost-cli

**Comment by gurpreetatwal** [comment]
Running into the same issue when upgrading from 2.13.1 to 3.29.1 

Node.js Version: 12.18.3

**Comment by sooxt98** [comment]
Is there a way to force migration for specific table? Or maybe reset specific table?

**Comment by gurpreetatwal** [comment]
I'm not too sure on the specifics of how ghost does the migrations, but I was able to work around the issue by going from 2.13.1 to 3.28.0 and then going from 2.13.1 to 3.28.1

Do you have a back up of the DB prior to upgrade? If so, I would just restore that and try out the 2 step upgrade

**Comment by sooxt98** [comment]
mysql  Ver 14.14 Distrib 5.7.31, for Linux (x86_64) using  EditLine wrapper
Ubuntu 18.04.3 LTS (GNU/Linux 4.15.0-112-generic x86_64)

**Comment by gurpreetatwal** [comment]
Apologies on not putting up enough info, I assumed that the bug wouldn't be OS dependent. 

Here's my info: 

- My machine's OS: `Ubuntu 18.04.5 LTS - 5.4.0-21-generic - x86_64`
- MySQL version: `5.7.31` - running in docker, `docker-compose.yaml` below
- I also face the same issue when running ghost in docker using `node:12`  as the base image

```yaml
mysql:
  image: mysql:5.7
  container_name: mysql
  environment:
    MYSQL_DATABASE: ghost
    MYSQL_ROOT_PASSWORD: development
  ports:
    - '3306:3306'
```

**Comment by ErisDS** [comment]
Critical part of the logs:

First, these are the 2.33 migrations

```
[2020-08-15 06:38:52] WARN Dropping table: members_stripe_customers
[2020-08-15 06:38:52] INFO Adding table: members_stripe_customers
[2020-08-15 06:38:52] INFO Adding table: members_stripe_customers_subscriptions
[2020-08-15 06:38:52] WARN Adding members_stripe_customers.email column
[2020-08-15 06:38:52] WARN Adding members_stripe_customers.name column
[2020-08-15 06:38:52] INFO Adding members.note column
```

Then the latest migrations

```
[2020-08-15 06:38:57] INFO No duplicate subscriptions found
[2020-08-15 06:38:57] INFO No duplicate customers found
[2020-08-15 06:38:57] INFO No orphaned customer records found
[2020-08-15 06:38:57] INFO No orphaned subscription records found
[2020-08-15 06:38:57] WARN Skipping ON DELETE CASCADE for "members_labels_member_id_foreign" constraint - already set
[2020-08-15 06:38:57] WARN Skipping ON DELETE CASCADE for "members_labels_label_id_foreign" constraint - already set
[2020-08-15 06:38:57] WARN Skipping "members_stripe_customers_member_id_foreign" foreign key constraint creation - already exists
[2020-08-15 06:38:57] WARN Skipping "members_stripe_customers_customer_id_unique" index creation - already exists
[2020-08-15 06:38:57] INFO Adding "members_stripe_customers_subscriptions_subscription_id_unique" index
[2020-08-15 06:38:57] INFO Removing tags.og_image column
```

Our code for detecting the index existing doesn't seem to work. 

Also note, the migrations kick off a rollback without any messaging!

**Comment by sooxt98** [comment]
![image](https://user-images.githubusercontent.com/13378059/90427187-cb8ed900-e0f4-11ea-8356-05de86938726.png)

@kevinansfield hi there what should i do if i deleted that `ghost_prod.members_stripe_customers_subscriptions` table, i'm stuck

**Comment by ErisDS** [comment]
@sooxt98 for support you need to use the [forum](https://forum.ghost.org).

**Comment by sooxt98** [comment]
hi @ErisDS I have solved my problem with `knex-migrator reset` and manual hand migration, thanks.

**Comment by daniellockyer** [comment]
Hey @sooxt98 & @gurpreetatwal :wave: We just released 3.29.2 with a fix for this, so please update using Ghost-CLI :slightly_smiling_face:
